### ISSUE TITLE
Inconsistent view.destroy() return values, isDestroyed() state

### ISSUE THREAD

**Issue by bazineta** [marionettejs/backbone.marionette#2345]
In LayoutView, we find the following:

```js
// Handle destroying regions, and then destroy the view itself.
    destroy: function() {
      if (this.isDestroyed) { return this; }
  
      this.regionManager.destroy();
      return Marionette.ItemView.prototype.destroy.apply(this, arguments);
    },
```

That is, if the view is already destroyed, the return value is 'this'.

The others however behave differently.  Here's CollectionView:

```js
 // Handle cleanup and other destroying needs for the collection of views
    destroy: function() {
      if (this.isDestroyed) { return; }
  
      this.triggerMethod('before:destroy:collection');
      this.destroyChildren();
      this.triggerMethod('destroy:collection');
  
      return Marionette.View.prototype.destroy.apply(this, arguments);
    },
```

Here, the return value is undefined if the view is already destroyed.

It seems to me that always returning 'this' from destroy() would be handy for chaining, for example, during BabySitter usage, when you could then immediately hand the return value to it's remove() function.

I'm not sure what if any value is derived from having inconsistency in the return values, so IMO it'd be preferable if they all acted as LayoutView does, but in any case, being consistent one way or the other would be preferable.

**Comment by jasonLaster** [comment]
nice find @bazineta. My guess is that you're the first person to run into this. @cmaher, @ahumphreys87 - have opinions?

@bazineta i'm curious what are you building?

**Comment by bazineta** [comment]
@jasonLaster It's a dashboard, drag and drop widgets, etc., using BabySitter for subview container management.  Quite mature at this point, but frankly, Marionette Inspector has shown me things that were hard to visualize otherwise, so I'm just about done on refactoring that's improved it greatly.  Inspector is just a fantastic tool.

**Comment by jasonLaster** [comment]
Awesome to hear!

**Comment by ahumphreys87** [comment]
Nice find @bazineta. Yeah I agree we should make this consistent and return this.

I think we do this in most places now to enable chaining.

@bazineta how would you feel about submitting a PR for this fix?

**Comment by cmaher** [comment]
Yep, `this` should definitely be returned

**Comment by samccone** [comment]
:+1: on this

**Comment by paulfalgout** [comment]
For that matter [`Marionette.Object.destroy`](https://github.com/marionettejs/backbone.marionette/blob/minor/src/object.js#L23) doesn't `return this;` either.

However, what sort of chaining should we be doing after a `destroy()`?  
maybe `myView.destroy()._ensureViewIsIntact()`

**Comment by ahumphreys87** [comment]
@paulfalgout valid point. My thoughts are that we should be consistent with whichever method we choose though, we cant really remove until v3.0 so I vote to add in where its missing and then maybe discuss removing in v3, thoughts?

**Comment by ahumphreys87** [comment]
Fixed in 2.4 via #2350
